pyupnp_async/upnp.py

```diff
@@ -92,11 +92,13 @@ def remaining(self):
     assert cp.start_time
 
     try:
+        resp = []
         async with timeout(cp.remaining, loop=loop):
             while not cp.timeout():
-                yield await cp.responses.get()
+                resp.append(await cp.responses.get())
                 if first_only:
-                    break
+                    return resp
+        return resp
     except asyncio.TimeoutError:
         pass
```

Comment:
In previous code, which used async generator, before a TimeoutError is raised, the caller will receive results (asynchronously). Then the TimeoutError will terminate the async generator.

But after changing to collect results first and then return the list, the results (list) should be returned even after catching a TimeoutError. Otherwise after timed out, the caller will get nothing even if there is a response. The initialization of `resp` should be moved outside the `try`...`except` and return from the `finally`. (Maybe just ignore any other exception for now?)

Reply:
You are absolutely right!

I was in such a hurry, I forgot to check the timeout.